File: 1063-best-sightseeing-pair/best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        
       m = values[-1] - (len(values)-1)
       ans = 0
       for i in range(len(values)-2,-1,-1):
           ans =max(ans,values[i] + i  + m)
           m = max(m,values[i]-i)
       return ans
              

File: 1063-best-sightseeing-pair/best-sightseeing-pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int m = values[0];
        int ans = 0;
        for(int i = 1;i<values.size();i++){
            ans = max(ans,m + values[i]-i);
            m= max(m,values[i]+i);
        }
        return ans;
    }
};